src/uct/cuda/cuda_ipc/cuda_ipc_md.c

```diff
@@ -313,18 +326,19 @@ uct_cuda_ipc_md_open(uct_component_t *component, const char *md_name,
     }
 
     /* Initially support caching accessibility of up to 16 other peers */
-    md->peer_accessible_cache = ucs_malloc(num_devices * md->uuid_map_capacity,
+    md->peer_accessible_cache = ucs_malloc(num_devices * md->uuid_map_capacity
+                                           * sizeof(ucs_ternary_value_t),
                                            "uct_cuda_ipc_peer_accessible_cache");
     if (md->peer_accessible_cache == NULL) {
         free(md->uuid_map);
         free(md);
         return UCS_ERR_NO_MEMORY;
     }
 
-    /* 0xFF = !cached, 1 = accessible, 0 = !accessible */
-    memset(md->peer_accessible_cache, 0xFF, num_devices * md->uuid_map_capacity);
-
-    ucs_spinlock_init(&md->peer_accessible_cache_lock, 0);
+    /* UCS_TRY = !cached, UCS_YES = accessible, UCS_NO = !accessible */
+    uct_cuda_ipc_memset(md->peer_accessible_cache, UCS_TRY,
+                        num_devices * md->uuid_map_capacity,
+                        sizeof(ucs_ternary_value_t));
```

Comment:
pass ucs_ternary_value_t* and length to this function
and rename it to: uct_cuda_ipc_accessible_cache_init
BTW maybe can initialize this cache to NULL/capacity=0, and let the uct_cuda_ipc_get_unique_index_for_uuid function allocate in on first use?

Reply:
@yosefe I've addressed the above two suggestions. I also ended up calling cuda_ipc cache insertion as part of reachability check. This saves us on one OpenMemHandle call and also to prevent a multi-threaded issue that I've documented. Lmk if the changes look ok.

Reply:
Also, I'm not sure why I'm seeing these errors on CI but not locally. 
```
In file included from /scrap/azure/agent-04/AZP_WORKSPACE/2/s/contrib/../src/uct/cuda/cuda_ipc/cuda_ipc_md.c:11:0:
/scrap/azure/agent-04/AZP_WORKSPACE/2/s/contrib/../src/uct/cuda/cuda_ipc/cuda_ipc_md.h:23:5: error: unknown type name 'ucs_ternary_value_t'
     ucs_ternary_value_t *peer_accessible_cache;
     ^
/scrap/azure/agent-04/AZP_WORKSPACE/2/s/contrib/../src/uct/cuda/cuda_ipc/cuda_ipc_md.c: In function 'uct_cuda_ipc_accessible_cache_init':
/scrap/azure/agent-04/AZP_WORKSPACE/2/s/contrib/../src/uct/cuda/cuda_ipc/cuda_ipc_md.c:77:28: error: 'ucs_ternary_value_t' undeclared (first use in this function)
     size_t offset = sizeof(ucs_ternary_value_t);
                            ^
/scrap/azure/agent-04/AZP_WORKSPACE/2/s/contrib/../src/uct/cuda/cuda_ipc/cuda_ipc_md.c:77:28: note: each undeclared identifier is reported only once for each function it appears in
/scrap/azure/agent-04/AZP_WORKSPACE/2/s/contrib/../src/uct/cuda/cuda_ipc/cuda_ipc_md.c:78:26: error: 'p' undeclared (first use in this function)
     ucs_ternary_value_t *p;
                          ^
/scrap/azure/agent-04/AZP_WORKSPACE/2/s/contrib/../src/uct/cuda/cuda_ipc/cuda_ipc_md.c:79:5: error: ISO C90 forbids mixed declarations and code [-Werror=declaration-after-statement]
     size_t i;
     ^
/scrap/azure/agent-04/AZP_WORKSPACE/2/s/contrib/../src/uct/cuda/cuda_ipc/cuda_ipc_md.c: In function 'uct_cuda_ipc_get_unique_index_for_uuid':
/scrap/azure/agent-04/AZP_WORKSPACE/2/s/contrib/../src/uct/cuda/cuda_ipc/cuda_ipc_md.c:110:40: error: 'ucs_ternary_value_t' undeclared (first use in this function)
                                 sizeof(ucs_ternary_value_t);
                                        ^
/scrap/azure/agent-04/AZP_WORKSPACE/2/s/contrib/../src/uct/cuda/cuda_ipc/cuda_ipc_md.c: In function 'uct_cuda_ipc_is_peer_accessible':
/scrap/azure/agent-04/AZP_WORKSPACE/2/s/contrib/../src/uct/cuda/cuda_ipc/cuda_ipc_md.c:146:5: error: unknown type name 'ucs_ternary_value_t'
     ucs_ternary_value_t *accessible;
     ^
cc1: all warnings being treated as errors
```

Reply:
@yosefe Remaining errors seem unrelated. Can you review the new changes when possible?